elytron/src/main/java/org/wildfly/extension/elytron/CredentialStoreAliasDefinition.java

```diff
@@ -215,16 +160,19 @@ protected void performRuntime(OperationContext context, ModelNode operation, Res
          */
         @Override
         public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
-            transformOperationAddress(operation);
+            if (!CASE_SENSITIVE.resolveModelAttribute(context, context.readResourceFromRoot(context.getCurrentAddress().getParent(), false).getModel()).asBoolean()) {
+                String aliasName = context.getCurrentAddressValue();
+                if (!aliasName.equals(aliasName.toLowerCase(Locale.ROOT))) {
+                    throw ElytronSubsystemMessages.ROOT_LOGGER.invalidAliasName(aliasName, context.getCurrentAddress().getParent().getLastElement().getValue());
+                }
+            }
             super.execute(context, operation);
         }
 
         @Override
         protected Resource createResource(OperationContext context, ModelNode operation) {
             Resource resource = Resource.Factory.create(true);
-            if (sameAlias(context, operation)) {
-                context.addResource(PathAddress.EMPTY_ADDRESS, resource);
-            }
+            context.addResource(PathAddress.EMPTY_ADDRESS, resource);
             return resource;
         }
```

Comment:
This doesn't need to be changed to merge this but I think this entire method override can be removed.